black-jack-redo.rb

```diff
@@ -121,65 +238,50 @@ def view_deck(array)
 	input = gets.chomp
 	input.downcase!      
 
-	if input == 'd' && cards == 52
-		#print_with_stars()
-		puts 'Your were dealt: '
-		user.push(array.pop)
-		user.push(array.pop)
-		puts array.length
-		puts user.length
-		#value = handcount(array)
-		#puts "Your cards add up to #{value} points!"
-		cards = 50
-		#format
-		puts '-------'
-		gameloop(array, user, cards)
-	elsif input == 'd' && cards != 52
-		#print_with_stars()
-		count4 = handcount(array)
-		ha, bool = drawone(array, cards, count4)
-		user.push(ha)
-		#user.push(drawone(array, cards, count4))
-		cards = cards - 1
-		#format
-		puts '-------'
+	if input == 'd' && array.length == 52
```

Comment:
Why does the array length matter?

Reply:
If the array length is less than 52 it means cards have already been dealt which triggers a draw rather than a new deal when pressing 'd'.